frontend/src/home/SwitchPlanPopper.tsx

```diff
@@ -0,0 +1,150 @@
+  addSchedule: (name: string, schedule: PastPresentSchedule) => void;
+  removeSchedule: (name: string, schedule: PastPresentSchedule) => void;
+}
+
+type Props = ReduxStoreSwitchSchedulesProps & ReduxDispatchSwitchSchedulesProps;
+
+interface SwitchSchedulePopperState {
+  anchorEl: null | HTMLElement;
+}
+
+export class SwitchPlanPopperComponent extends React.Component<
+  Props,
+  SwitchSchedulePopperState
+> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      anchorEl: null,
+    };
+  }
+
+  /**
+   * Toggles opening the SwitchPlanPopper when the plan name is clicked.
+   * @param event mouse event trigger
+   */
+  handleClick(event: React.MouseEvent<HTMLElement>) {
+    if (this.state.anchorEl === null) {
+      this.setState({
+        anchorEl: event.currentTarget,
+      });
+    } else {
+      this.setState({
+        anchorEl: null,
+      });
+    }
+  }
+
+  /**
+   * Updates this user's active schedule based on the schedule selected in the dropdown.
+   */
+  onChoosePlan(value: string) {
+    const newSchedule = this.props.schedules.find(s => s.name === value);
+    if (newSchedule) {
+      const newActive = this.props.schedules.indexOf(newSchedule);
+      this.props.setActiveSchedule(newActive);
+      this.setState({
+        anchorEl: null,
+      });
+    }
+  }
+
+  /**
+   * Enables hiding the popper by clicking anywhere else on the screen.
+   */
+  handleClickAway() {
+    this.setState({
+      anchorEl: null,
+    });
+  }
+
+  render() {
+    return (
+      <div>
+        <SwitchPlanContainer onClick={event => this.handleClick(event)}>
+          <h2>{`- ${this.props.activeSchedule.name}`}</h2>
+          {Boolean(this.state.anchorEl) ? (
+            <KeyboardArrowUpIcon />
+          ) : (
+            <KeyboardArrowDownIcon />
+          )}
+        </SwitchPlanContainer>
+        <SwitchPlanMenu
+          id={"simple-popper"}
+          open={Boolean(this.state.anchorEl)}
+          anchorEl={this.state.anchorEl}
+          onClose={() => this.handleClickAway()}
+          anchorOrigin={{
+            vertical: "bottom",
+            horizontal: "left",
+          }}
+        >
+          {this.props.schedules.map(s => (
+            <MenuItem onClick={() => this.onChoosePlan(s.name)} key={s.name}>
+              {s.name}
+            </MenuItem>
+          ))}
+        </SwitchPlanMenu>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = (state: AppState) => ({
+  schedules: getSchedulesFromState(state),
+  activeSchedule: getActiveScheduleFromState(state),
+});
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  setActiveSchedule: (activeSchedule: number) =>
+    dispatch(setActiveScheduleAction(activeSchedule)),
+  addSchedule: (name: string, schedule: PastPresentSchedule) =>
+    dispatch(addScheduleAction(schedule, name)),
+  removeSchedule: (name: string, schedule: PastPresentSchedule) =>
+    dispatch(removeScheduleAction(schedule, name)),
```

Comment:
I don't think you use removeSchedule and addSchedule in here, so it's probably not necessary to keep these fields